infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -149,7 +152,7 @@ public ManagementServer(ServerContext serverContext) {
                 corfuRuntime, serverContext, clusterContext, failureDetector,managementLayout
         );
 
-        orchestrator = new Orchestrator(corfuRuntime, serverContext);
+        orchestrator = new Orchestrator(corfuRuntime, serverContext, streamLog);
```

Comment:
There are a couple of reasons. 

1) Orchestrator is getting scheduled on the same node as the log unit for the three workflows that use the state transfer and restore redundancy (locally). 
2) State Transfer is invoked in three cases and in two of these cases it needs an instance of the local reset/new stream log for writing data.
3) Writing data during the state transfer is happening to the segments of the stream log that are not open to the client, so concurrent writes to the same parts of the stream log do not happen. 
4) State Transfer is effectively stateless just like the workflows that invoke it. Since now we have only one hop (We don't schedule an orchestrator on the different node than the one that needs a transfer), to restore the lost state we can scan the actual part of a stream log directly without having to write RPC calls to the local log unit to check if we are missing this data. 

However, I do agree that passing a stream log directly to the orchestrator is not a good idea. 
We need to create a state transfer manager instance for the orchestrator to use during the workflows.